roles/update_etc_hosts/tasks/update_host_present.yml

```diff
@@ -0,0 +1,181 @@
+    that: not( ( thishost.node_name is undefined) or ( thishost.node_name is none) or ( thishost.node_name | trim == '') )
+    msg: "Variable 'node_name' is undefined or empty. Please define it your host list"
+
+- name: Ensure node_domain is set
+  ansible.builtin.set_fact:
+    __update_etc_hosts_domain: "{{ thishost.node_domain | default(sap_domain) | default(ansible_domain) }}"
+
+# Necessary, if defaults are both undefined
+- name: Verify that variable domain_name is set
+  ansible.builtin.assert:
+    that: >
+      not( ( __update_etc_hosts_domain is undefined) or
+      ( __update_etc_hosts_domain is none) or
+      ( __update_etc_hosts_domain | trim == '') )
+    msg: "Variable 'domain_name' is undefined or empty. Please define it your host list"
+
+- name: Set default values
+  ansible.builtin.set_fact:
+    __update_etc_hosts_comment: "{{ (thishost.node_comment | d('') + ' ' + thishost.hana_site | d('')) | trim }}"
+    __update_etc_hosts_alias_mode: "{{ thishost.alias_mode | default('merge') }}"
+
+- name: Prepend Hashtag to comment
+  when: __update_etc_hosts_comment|length > 0
+  ansible.builtin.set_fact:
+    __update_etc_hosts_comment: "# {{ __update_etc_hosts_comment }}"
+
+# The following block reads the existing aliases of a host from /etc/hosts
+# and merges it with the defined aliases in the struct
+#
+# 1. select the line, where the first entry is the ip-adress thishost.node_ip
+# 2. loop over all hostname entries in the selected line (2 bis NF=last element in line)
+# 3. stop looping when a comment sign is found (because these are comments)
+# 4. print an element, if it is not the hostname or FQDN we want to add
+#
+# => __update_etc_hosts_register_aliases.stdout contains a list of aliases of thishost.node_ip
+#
+# trunk-ignore(checkov/CKV2_ANSIBLE_3)
+- name: Merge existing aliases with new alis list
+  when: __update_etc_hosts_alias_mode != "overwrite"
+  block:
+    - name: Get all existing hostname aliases of {{ thishost.node_ip }}
+      ansible.builtin.shell: |
+        awk '( $1 == "{{ thishost.node_ip }}" ) {
+          for (i=2; i<=NF; ++i) {
+            if ( $i == "#" ) { break }
+            if (( $i != "{{ thishost.node_name }}" ) && ( $i != "{{ thishost.node_name }}.{{ __update_etc_hosts_domain }}" )) { printf " "$i }
+          }
+        }' "{{ __update_etc_hosts_file }}"
+      register: __update_etc_hosts_register_aliases
+      changed_when: false
+
+    - name: Add defined aliases
+      ansible.builtin.set_fact:
+        __update_etc_hosts_aliases: "{{ (__update_etc_hosts_register_aliases.stdout.split(' ') + thishost.aliases | d([])) | unique | join(' ') }}"
+
+- name: Overwrite existing aliases
+  when: __update_etc_hosts_alias_mode == "overwrite"
+  ansible.builtin.set_fact:
+    __update_etc_hosts_aliases: "{{ thishost.node_aliases | d([]) | unique | join(' ') }}"
+
+- name: Display host and domain name, and IP address before the modification
+  ansible.builtin.debug:
+    msg:
```

Comment:
I suggest to surround the variable contents with a single quote to make it easier to see the string boundaries. Might be useful during problem analysis. Example:
      - "aliases  = '{{ __update_etc_hosts_aliases }}'"

Reply:
done